borg/helpers.py

```diff
@@ -376,6 +375,27 @@ def _match(self, path):
         return (self.regex.match(path + os.path.sep) is not None)
 
 
+class ShellPattern(PatternBase):
+    """Shell glob patterns to exclude.  A trailing slash means to
+    exclude the contents of a directory, but not the directory itself.
+    """
+    PREFIX = "sh"
+
+    def _prepare(self, pattern):
+        sep = os.path.sep
+
+        if pattern.endswith(sep):
+            pattern = os.path.normpath(pattern).rstrip(sep) + sep + "**" + sep + "*" + sep
+        else:
+            pattern = os.path.normpath(pattern) + sep + "**" + sep + "*"
```

Comment:
hmm, have to look at the other code, but iirc it had hardcoded "**/" (not "**<sep>").